agent/run.sh

```diff
@@ -29,6 +29,11 @@ export CATTLE_HOME=${CATTLE_HOME:-/var/lib/cattle}
 
 check_debug
 
+RUNNING_IMAGE="$(docker inspect $(hostname)| jq -r .[].Config.Image)"
+if [[ ${RUNNING_IMAGE} != ${RANCHER_AGENT_IMAGE} ]]; then
```

Comment:
As a safety measure you might want to do

`if [[ -n ${RUNNING_IMAGE} && ${RUNNING_IMAGE} != ${RANCHER_AGENT_IMAGE} ]]; then`